### PR TITLE
Updated User to Have Role

### PR DESCRIPTION
**Pull request by KingDad**
Added role column to user table and a test to user_spec.rb to check that the role attribute is present and defaults to member

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,4 +2,5 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
```
Comment (prophetsfall) [inline app/controllers/application_controller.rb]: Please remove extra line

File: app/controllers/trains_controller.rb [hunk 2]
```diff
@@ -8,4 +9,12 @@ def show
     @train = Train.find(params[:id])
   end
 
+  protected
+
+  def authorize_user
+    if !user_signed_in? || !current_user.admin?
+      flash[:notice] = "You do not have access to this page"
+    end
+  end
```
Comment (prophetsfall) [inline app/controllers/trains_controller.rb:18]: Nice method.

**Review by prophetsfall — CHANGES_REQUESTED** [review]
Please make the one minor change and push your change up. Let me know when you've done that and I'll merge.

File: config/routes.rb [hunk 3]
```diff
@@ -1,9 +1,8 @@
 Rails.application.routes.draw do
-  root 'homes#index'
+  root 'trains#index'
   devise_for :users
 
-  resources :trains, only: [:index, :show]
+  resources :trains
```
Comment (CassandraNewell) [inline config/routes.rb:5]: Please restrict this to only the methods you need (using `only:`)

**Review by CassandraNewell — CHANGES_REQUESTED** [review]

**Pull request merged**